File: lib/Karve/CustomControls/DataGrid/Clases/DataGridColumn/DataGridColumn.vb
Imports Telerik.WinControls.UI



Public Class DataGridColumn
    Inherits GridViewDataColumn

    Enum eTipoColumn
        Pk
        Fecha
        Hora
        Texto
        Observaciones
        Numerico
    End Enum

    Dim _Item As Integer
    Dim _Campo As String
    Dim _Tabla As String
    Dim _ExpresionBd As String
    Dim _BackColor As System.Drawing.Color = Drawing.Color.White
    Dim _eTipoColumn As eTipoColumn

    Property Item As Integer
        Get
            Return _Item
        End Get
        Set(value As Integer)
            _Item = value
        End Set
    End Property

    Property Campo As String
        Get
            Return _Campo
        End Get
        Set(value As String)
            _Campo = value
        End Set
    End Property

    Property ExpresionBd As String
        Get
            Return _ExpresionBd
        End Get
        Set(value As String)
            _ExpresionBd = value
        End Set
    End Property

    Property Tabla As String
        Get
            Return _Tabla
        End Get
        Set(value As String)
            _Tabla = value
        End Set
    End Property

    Property AliasCampo As String
        Get
            Return FieldName
        End Get
        Set(value As String)
            FieldName = value
        End Set
    End Property

    Property BackColor As System.Drawing.Color
        Get
            Return _BackColor
        End Get
        Set(value As System.Drawing.Color)
            _BackColor = value
        End Set
    End Property

    Property TipoColumna As eTipoColumn
        Get
            Return _eTipoColumn
        End Get
        Set(value As eTipoColumn)
            _eTipoColumn = value
            EstablecePropiedadesTipoColumna()
        End Set
    End Property

    Private Sub EstablecePropiedadesTipoColumna()
        Select Case _eTipoColumn
            Case eTipoColumn.Pk : ColumnaPk()
            Case eTipoColumn.Fecha : ColumnaFecha()
        End Select

    End Sub

    Private Sub ColumnaPk()
        _BackColor = Drawing.Color.SkyBlue
        MyBase.Width = 80
    End Sub

    Private Sub ColumnaFecha()
        Dim sAlias As String = _Tabla
        If sAlias <> "" Then sAlias = sAlias & "."
        _ExpresionBd = " DATEFORMAT(" & sAlias & MyBase.FieldName & ", 'dd/mm/yyyy') "
        MyBase.Width = 100
        MyBase.AllowResize = False
    End Sub

End Class

File: src/Controls/KarveControls/DataGrid/DataGridHelpers/DataGridColumns/DataGridColumn.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
using System.Windows.Media;
using Telerik.WinControls.UI;

namespace KarveControls
{
    public class DataGridColumn : GridViewDataColumn
    {

        public enum eTipoColumn
        {
            Pk,
            Fecha,
            Hora,
            Texto,
            Observaciones,
            Numerico
        }

        int _Item;
        string _Campo;
        string _Tabla;
        string _ExpresionBd;
        System.Drawing.Color _BackColor = System.Drawing.Color.White;

        eTipoColumn _eTipoColumn;

        public int Item
        {
            get { return _Item; }
            set { _Item = value; }
        }

        public string Campo
        {
            get { return _Campo; }
            set { _Campo = value; }
        }

        public string ExpresionBd
        {
            get { return _ExpresionBd; }
            set { _ExpresionBd = value; }
        }

        public string Tabla
        {
            get { return _Tabla; }
            set { _Tabla = value; }
        }

        public string AliasCampo
        {
            get { return FieldName; }
            set { FieldName = value; }
        }

        public System.Drawing.Color BackColor
        {
            get { return _BackColor; }
            set { _BackColor = value; }
        }

        public eTipoColumn TipoColumna
        {
            get { return _eTipoColumn; }
            set
            {
                _eTipoColumn = value;
                EstablecePropiedadesTipoColumna();
            }
        }

        private void EstablecePropiedadesTipoColumna()
        {
            switch (_eTipoColumn)
            {
                case eTipoColumn.Pk:
                    ColumnaPk();
                    break;
                case eTipoColumn.Fecha:
                    ColumnaFecha();
                    break;
            }

        }

        private void ColumnaPk()
        {
//            _BackColor = Drawing.Color.SkyBlue;
            base.Width = 80;
        }

        private void ColumnaFecha()
        {
            string sAlias = _Tabla;
            if (!string.IsNullOrEmpty(sAlias))
                sAlias = sAlias + ".";
            _ExpresionBd = " DATEFORMAT(" + sAlias + base.FieldName + ", 'dd/mm/yyyy') ";
            base.Width = 100;
            base.AllowResize = false;
        }

    }
}